### PR TITLE
Render image details for media gallery grid

### PR DESCRIPTION
**Pull request by Nazar65**
### Fixed Issues (if relevant)

1. magento/adobe-stock-integration#829: Render image names for media gallery grid
2. ...

### Manual testing scenarios (*)

1. Login to Admin Panel
2. Select "Media Gallery Development" from the menu on the right

Images data like dimension name content_type added under the images
DeepinScreenshot_select-area_20191224153506

DeepinScreenshot_select-area_20191224153617

### CODE REVIEW

File: MediaGalleryUi/view/adminhtml/web/template/grid/masonry.html [hunk 1]
```diff
@@ -0,0 +1,17 @@
+<!--
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+-->
+<div data-role="grid-wrapper" class="masonry-image-grid" attr="'data-id': containerId">
+    <div class="masonry-image-column media-gallery-image-column" repeat="foreach: rows, item: '$row'">
```
Comment (sivaschenko) [inline MediaGalleryUi/view/adminhtml/web/template/grid/masonry.html]: I don't think it's necessary to override the masonry component and template only to add the class. I'd propose to use '.media_gallery-index-index' parent selector instead for now, and later (when media gallery will be moved to slide panel) adjust it to slide panel selector

File: MediaGalleryUi/view/adminhtml/web/js/grid/columns/image.js [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+define([
+    'jquery',
+    'Magento_Ui/js/grid/columns/image'
+], function ($, image) {
+    'use strict';
+
+    return image.extend({
```
Comment (sivaschenko) [inline MediaGalleryUi/view/adminhtml/web/js/grid/columns/image.js]: Is it possible to override only the `bodyTmpl` and prepare the data for rendering on the backend when populating `media_gallery_asset_grid` table?

File: MediaGalleryUi/view/adminhtml/web/js/grid/columns/image.js [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+        },
+
+        /**
+         * Returns dimensions to given record.
+         *
+         * @param {Object} record
+         * @return {String}
+         */
+        getDimensions: function (record) {
+            return record.width + 'x' + record.height;
```
Comment (sivaschenko) [inline MediaGalleryUi/view/adminhtml/web/js/grid/columns/image.js]: Can be concatenated in the template

File: MediaGalleryUi/view/adminhtml/web/js/grid/columns/image.js [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+        },
+
+        /**
+         * Display source if image was downloaded in adobe stock.
+         *
+         * @param {Object} record
+         * @return {String}
+         */
+        getSource: function (record) {
+            return record.source ? this.adobeStockIconUrl : false
```
Comment (sivaschenko) [inline MediaGalleryUi/view/adminhtml/web/js/grid/columns/image.js]: The source may be not only the Adobe Stock. I think we should simply have a source_icon_url as a grid column. A class on the backend like `\Magento\MediaGalleryUi\Ui\Component\Listing\Columns\Url` may be responsible for preparing such urls

File: MediaGalleryUi/view/adminhtml/web/template/grid/columns/image.html [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+-->
+<div class="media-gallery-image">
+    <div class="masonry-image-block media-gallery-image-block" attr="'data-id': $col.getId($row())">
+        <img attr="src: $col.getUrl($row())" class="media-gallery-image-column" css="$col.getClasses($row())" click="function(){ expandPreview($row()) }" data-role="thumbnail"/>
+    </div>
+</div>
+<div class="masonry-image-description">
+    <ul class="media-gallery-image-details">
+        <li class="name" data-bind="text: $col.getName($row())"></li>
+        <li class="source"><img if="$col.getSource($row())" class="adobe-stock-icon" data-bind="attr: {src: $col.getSource($row())}" /></li>
```
Comment (sivaschenko) [inline MediaGalleryUi/view/adminhtml/web/template/grid/columns/image.html]: See my comment above, also, I think we can use a condition like `ko if: $row().source_icon_url ` to display source icon only if it is specified

**Review by sivaschenko — CHANGES_REQUESTED** [review]
Hi @Nazar65 thanks for the pull request! It looks amazing! However, I have a couple of propositions regarding minimizing the overriding, please see my comments below!

File: MediaGalleryUi/view/adminhtml/web/template/grid/masonry.html (hunk 1 above)
Comment (Nazar65) [reply MediaGalleryUi/view/adminhtml/web/template/grid/masonry.html]: ok

File: MediaGalleryUi/view/adminhtml/web/js/grid/columns/image.js (hunk 4 above)
Comment (Nazar65) [reply MediaGalleryUi/view/adminhtml/web/js/grid/columns/image.js]: ok

File: MediaGalleryUi/view/adminhtml/web/template/grid/columns/image.html (hunk 5 above)
Comment (Nazar65) [reply MediaGalleryUi/view/adminhtml/web/template/grid/columns/image.html]: ok

File: MediaGalleryUi/view/adminhtml/web/js/grid/columns/image.js (hunk 3 above)
Comment (Nazar65) [reply MediaGalleryUi/view/adminhtml/web/js/grid/columns/image.js]: ok

File: MediaGalleryUi/view/adminhtml/web/js/grid/columns/image.js (hunk 2 above)
Comment (Nazar65) [reply MediaGalleryUi/view/adminhtml/web/js/grid/columns/image.js]: yes :+1:

**Review by sivaschenko — APPROVED** [review]

**Pull request merged**